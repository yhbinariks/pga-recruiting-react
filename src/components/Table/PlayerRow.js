import React from 'react';
import PropTypes from 'prop-types';

const PlayerRow = ({ player, onEdit, onDelete }) => (
  <tr>
    <td>{`${player.lastName}, ${player.firstName}`}</td>
    <td>{player.score}</td>
    <td>
      <button
        type="button"
        className="table__action-btn"
        onClick={onEdit('edit', player.id)}
      >
        Edit
      </button>
      <button
        type="button"
        className="table__action-btn"
        onClick={onDelete(player.id)}
      >
        Delete
      </button>
    </td>
  </tr>
);

const propTypes = {
  onEdit: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  player: PropTypes.shape({
    lastName: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
  }).isRequired,
};

PlayerRow.propTypes = propTypes;

export default PlayerRow;
