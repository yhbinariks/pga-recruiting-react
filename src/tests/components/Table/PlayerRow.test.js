import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import PlayerRow from '../../../components/Table/PlayerRow';

describe('PlayerRow component', () => {
  it('snapshot', () => {
    const props = {
      player: {
        firstName: 'petro3',
        lastName: 'chu3',
        score: 32,
        id: '1233',
      },
      onEdit: jest.fn(),
      onDelete: jest.fn(),
    };
    const playerRow = shallow(
      <table>
        <tbody>
          <PlayerRow {...props} />
        </tbody>
      </table>,
    );
    expect(toJson(playerRow)).toMatchSnapshot();
  });

  it('edit click', () => {
    const props = {
      player: {
        firstName: 'petro3',
        lastName: 'chu3',
        score: 32,
        id: '1233',
      },
      onEdit: jest.fn(),
      onDelete: jest.fn(),
    };
    const playerRow = mount(
      <table>
        <tbody>
          <PlayerRow {...props} />
        </tbody>
      </table>,
    );
    playerRow.find('button').at(0).simulate('click');

    expect(props.onEdit).toHaveBeenCalledWith('edit', props.player.id);
  });
  it('delete click', () => {
    const props = {
      player: {
        firstName: 'petro3',
        lastName: 'chu3',
        score: 32,
        id: '1233',
      },
      onEdit: jest.fn(),
      onDelete: jest.fn(),
    };
    const playerRow = mount(
      <table>
        <tbody>
          <PlayerRow {...props} />
        </tbody>
      </table>,
    );
    playerRow.find('button').at(1).simulate('click');

    expect(props.onDelete).toHaveBeenCalledWith(props.player.id);
  });
});
