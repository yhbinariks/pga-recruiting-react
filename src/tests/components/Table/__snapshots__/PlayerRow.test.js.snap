// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerRow component snapshot 1`] = `
<table>
  <tbody>
    <PlayerRow
      onDelete={[MockFunction]}
      onEdit={[MockFunction]}
      player={
        Object {
          "firstName": "petro3",
          "id": "1233",
          "lastName": "chu3",
          "score": 32,
        }
      }
    />
  </tbody>
</table>
`;
